### ISSUE TITLE
Fontconfig issue?

### ISSUE THREAD

**Issue by Crissium** [garglk/garglk#827]
Somehow gargoyle seems to have suddenly broken on my machine (Debian bookworm). My memory is a bit foggy but I think it stopped working only yesterday. Whenever I run a game I'd get:
```bash
gargoyle-free r1.z3 
bocfel: fccache.c:808: FcCacheFini: Assertion `fcCacheChains[i] == NULL' failed.
```

If I just start `gargoyle-free` with no arguments, the file selection dialogue appears, but when I select one, it crashes again with the error message shown above.

It seems to be a fontconfig issue, and I googled around but found no effective solution. I tried `fc-cache -r` or `-f` and the assertion would still fail.

Is there any more information needed? Should I report it to fontconfig instead?

Edit:

I checked the dpkg log and found no related package update. Nor have I touched the fonts recently.

No other application has this issue.

**Comment by cspiegel** [comment]
This is, unfortunately, a bug in 2022.1 that's been fixed in 2023.1.  If you're comfortable building Gargoyle, you can grab the 2023.1 source and build that, but of course it's not super user friendly, that way.  I'm not too familiar with Debian, but they _do_ provide a 2023.1 build [here](https://packages.debian.org/unstable/gargoyle-free) for stable and testing. If Debian allows newer packages to be installed on older releases, then this may be a solution.

If the Debian maintainers are interested in backporting the fix, it was done in commit 524a1b6ad.  It doesn't cleanly apply to 2022.1, but it's a small enough patch that it ought to be simple to do it by handl

**Comment by Crissium** [comment]
Thanks! I checked out the 2022.1 version and applied the patch by hand, and it doesn't complain about that failed assertion any more. I've already asked the maintainers to backport this. (Debian stable doesn't have libjpeg>=8.0, so the latest version cannot be built.)
